### PR TITLE
Create greatest_common_divisor.go

### PR DESCRIPTION
**Pull request by vedantmamgain**

### CODE REVIEW

File: math/greatest_common_divisor.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+func gcd(a, b int) int {
+	if b == 0 {
+		return a
+	} else {
+		return gcd(b, a%b)
+	}
+}
+
+func main() {
+	print(fastExponentiation(3, 0))
```
Comment (nomisrevol) [inline math/greatest_common_divisor.go:19]: Hm, what's that?

Comment (vedantmamgain) [reply math/greatest_common_divisor.go]: My bad !! Fixed it

**Review by nomisrevol — CHANGES_REQUESTED** [review]

File: math/greatest_common_divisor.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+
+package main
+
+import ("fmt")
+
+func gcd(a, b int) int {
+	if b == 0 {
+		return a
+	} else {
```
[the remark this answers is not in the document]
Comment (vedantmamgain) [reply math/greatest_common_divisor.go]: Done .

File: math/greatest_common_divisor.go [hunk 3]
```diff
@@ -1,22 +1,15 @@
...
 import ("fmt")
 
 func gcd(a, b int) int {
 	if b == 0 {
 		return a
-	} else {
-		return gcd(b, a%b)
-	}
+	} 
+   return gcd(b, a%b)
```
Comment (nomisrevol) [inline math/greatest_common_divisor.go:9]: It seems wrong indentation. Did you use golint or some lint tool to format it?

**Review by nomisrevol — CHANGES_REQUESTED** [review]

File: math/greatest_common_divisor.go (hunk 3 above)
Comment (vedantmamgain) [reply math/greatest_common_divisor.go]: Now I've fixed the indentation with the linter.

**Review by nomisrevol — APPROVED** [review]
LGTM

**Comment by nomisrevol** [comment]
Fix the lint, squash into 1 single commit, then I will merge this PR.

**Comment by vedantmamgain** [comment]
Squashed all the commits into one.

**Review by vedantmamgain — COMMENTED** [review]
Squashed all the commits check it now.

**Pull request merged**